auto_dev/data/repo/templates/autonomy/install.sh

```diff
@@ -38,82 +35,97 @@ get_download_url() {
             protolint_arch="arm64"
             ;;
         *)
-            echo "Unsupported architecture: $arch" >&2
+            echo "Unsupported architecture: ${arch}" >&2
             return 1
             ;;
     esac
 
-    if [ "$tool" = "protoc" ]; then
-        if [ "$os" = "Darwin" ]; then
+    if [ "${tool}" = "protoc" ]; then
+        if [ "${os}" = "Darwin" ]; then
             echo "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-osx-universal_binary.zip"
         else
             echo "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-${protoc_os}-${protoc_arch}.zip"
         fi
-    elif [ "$tool" = "protolint" ]; then
+    elif [ "${tool}" = "protolint" ]; then
         echo "https://github.com/yoheimuta/protolint/releases/download/v${PROTOLINT_VERSION}/protolint_${PROTOLINT_VERSION}_${protolint_os}_${protolint_arch}.tar.gz"
     else
-        echo "Unknown tool: $tool" >&2
+        echo "Unknown tool: ${tool}" >&2
         return 1
     fi
 }
 
 # Function to download and install a tool
 install_tool() {
-    local tool=$1
+    local tool="$1"
     local url
-    local venv_dir=$(echo $(poetry run which python) | sed 's|\(.*\)/python|\1|')
-    url=$(get_download_url "$tool") || return 1
+    local venv_dir
+
+    # Get virtualenv path from poetry env info
+    printf "Getting virtualenv path from poetry... "
+    venv_dir="$(poetry env info -p)/bin"
+    if [ $? -ne 0 ] || [ -z "${venv_dir}" ]; then
+        echo "Error: Could not find poetry virtualenv" >&2
+        return 1
+    fi
+    echo "${venv_dir}"
+
+    if [ ! -d "${venv_dir}" ]; then
+        echo "Error: Virtualenv directory ${venv_dir} does not exist" >&2
+        return 1
+    fi
 
-    echo "Installing $tool"
+    url="$(get_download_url "${tool}")" || return 1
+
+    echo "Installing ${tool}"
     local temp_dir
-    temp_dir=$(mktemp -d)
-    cd "$temp_dir"
+    temp_dir="$(mktemp -d)"
+    cd "${temp_dir}" || exit 1
 
-    if ! wget -q "$url"; then
-        echo "Failed to download $tool" >&2
-        cd - > /dev/null
-        rm -rf "$temp_dir"
+    if ! wget -q "${url}"; then
+        echo "Failed to download ${tool}" >&2
+        cd - > /dev/null || exit 1
+        rm -rf "${temp_dir}"
         return 1
     fi
 
     local filename
-    filename=$(basename "$url")
+    filename="$(basename "${url}")"
 
-    case "$filename" in
+    case "${filename}" in
```

Comment:
why change all of this? :)

Reply:
the idea was to make the install script shell-agnostic to work with zsh. I was trying to run it manually and getting errors before realizing it only works with bash. that commit is here which I can just revert 4f8f0c84afd248e5662ccb0023285f17f5db9fb8

Reply:
reverted formatting changes in 79afc73e5c86ed64a8afa8c4da1870b7ba80c036